src/views/Lottery/components/BuyTicketsModal/BuyTicketsModal.tsx

```diff
@@ -81,7 +82,8 @@ const BuyTicketsModal: React.FC<BuyTicketsModalProps> = ({ onDismiss }) => {
   const [maxTicketPurchaseExceeded, setMaxTicketPurchaseExceeded] = useState(false)
   const [userNotEnoughCake, setUserNotEnoughCake] = useState(false)
   const lotteryContract = useLotteryV2Contract()
-  const cakeContract = useCake()
+  const cakeContractReader = useCake(false)
+  const cakeContractApprover = useCake()
```

Comment:
Could you help me understand why we need to split into 2 instances ? What is the problem with using 1 instance ?

Reply:
When you get a contract with signer, it uses rpc nodes defined by user’s wallet when making request. It might be sometimes laggy and returns slow response which affects ux.

Reply:
Got it. Thanks for your explanation